components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -149,6 +160,40 @@ public static Map<String, String> getPlaceholderValues(EmailTemplate emailTempla
 
         Map<String, String> configFilePlaceholders = getConfigFilePlaceholders();
 
+        JsonNode brandingPreferences = null;
+        Map<String, String> brandingFallbacks = getBrandingFallbacksFromConfigFile();
+
+        if (Boolean.parseBoolean(
+                IdentityUtil.getProperty(NotificationConstants.EmailNotification.ENABLE_ORGANIZATION_LEVEL_EMAIL_BRANDING))) {
+            try {
+                BrandingPreferenceManager brandingPreferenceManager = new BrandingPreferenceManagerImpl();
+                BrandingPreference responseDTO = brandingPreferenceManager.getBrandingPreference(
+                        BrandingPreferenceMgtConstants.ORGANIZATION_TYPE,
+                        placeHolderData.get("tenant-domain"),
+                        BrandingPreferenceMgtConstants.DEFAULT_LOCALE);
+
+                ObjectMapper objectMapper = new ObjectMapper();
+                String json = objectMapper.writeValueAsString(responseDTO.getPreference());
+                brandingPreferences = objectMapper.readTree(json);
+
+                if (!brandingPreferences.at(NotificationConstants.EmailNotification.BRANDING_PREFERENCES_IS_ENABLED_PATH)
+                        .asBoolean()) {
+                    brandingPreferences = null;
+                }
+            } catch (BrandingPreferenceMgtException e) {
+                if (BrandingPreferenceMgtConstants.ErrorMessages.ERROR_CODE_BRANDING_PREFERENCE_NOT_EXISTS.getCode()
+                        .equals(e.getErrorCode())) {
+                    brandingPreferences = null;
+                } else {
+                    String message = "Error occurred while retrieving branding preferences for organization " + placeHolderData.get("tenant-domain");
```

Comment:
Add a check for debug enabled

Reply:
Fixed with 5072ba8688399a37941c8972ace49c2dfe067ada